### ISSUE TITLE
null-ls integration: sticky errors

### ISSUE THREAD

**Issue by gravis** [ray-x/go.nvim#322]
Hi, I'm trying to configure the null-[ls integration](https://github.com/ray-x/go.nvim#integrate-null-ls), and got it to work, but errors are not going away when a test if fixed.

By the way, I think there's an error in the code sample provided:

```lua
table.insert(sources, gotest, golangci_lint)
table.insert(sources, gotest_codeaction)
```

should be 

```lua
table.insert(sources, gotest)
table.insert(golangci_lint)
table.insert(sources, gotest_codeaction)
```

According to https://www.lua.org/pil/19.2.html, `gotest` is considered as a position in this case, and the declaration will fail. Could you please confirm? Thanks

**Comment by gravis** [comment]
I forgot to mention I'm using [LazyVim](https://github.com/LazyVim/LazyVim) (in case it could be related with that instead).

**Comment by gravis** [comment]
By the way, I'm also running into this: https://github.com/golangci/golangci-lint/issues/536
golangci-lint is apparently running on the current file only, if I understand this correctly: https://github.com/ray-x/go.nvim/blob/1be8ef7cdf2dd355b83f330b2e6861a992fcc432/lua/go/null_ls.lua#L200

**Comment by ray-x** [comment]
Yes, it is meaningful to have it run on the active editing buffer.  For any decent project, run a full lint scan each time on buffer save will just burn some CPU and memory.

**Comment by howiieyu** [comment]
> Yes, it is meaningful to have it run on the active editing buffer. For any decent project, run a full lint scan each time on buffer save will just burn some CPU and memory.

I think the better solution is to run it on the directory (package) where the current file is located; this can avoid the 'undefined' error diagnostic message, and not much CPU and memory will be consumed.
